components/admin/requests/processing/TasksCard.tsx

```diff
@@ -201,98 +211,131 @@ export default function ProcessingTasksCard({ applicationId }: ProcessingTasksCa
             applicationId={applicationId}
             requestType={'Request'}
-            onConfirmed={() => setHasReviewedRequest(true)}
+            onConfirmed={() => setReviewRequestTask(true)}
           >
-            {invoiceNumber === null ? (
+            {!reviewRequestTask ? (
               <Button
                 marginLeft="auto"
                 height="35px"
                 bg="background.gray"
                 _hover={{ bg: 'background.grayHover' }}
                 color="black"
+                disabled={!appNumber || !appHolepunched || !walletCardCreated}
               >
-                <Text textStyle="xsmall-medium">Review Information</Text>
+                <Text textStyle="xsmall-medium">Review information</Text>
               </Button>
             ) : (
-              <Button variant="ghost" textDecoration="underline black">
+              <Button
+                variant="ghost"
+                textDecoration="underline black"
+                onClick={() => {
+                  setDocumentUploadTask(false);
+                  setGenerateInvoiceTask(false);
+                  handleMailOut(false);
+                  setReviewRequestTask(false);
+                }}
+              >
                 <Text textStyle="caption" color="black">
-                  Edit number
+                  Undo Review
                 </Text>
               </Button>
             )}
           </ReviewInformationModal>
         </ProcessingTaskStep>
-        {/* Task 5: Assign invoice number: Assign number (MODAL) */}
+        {/* Task 5: Generate Invoice */}
         <ProcessingTaskStep
           id={5}
-          label={`Assign invoice number${invoiceNumber === null ? '' : `: ${invoiceNumber}`}`}
-          description="Include permit number, expiry date, full name, and birth month"
-          isCompleted={invoiceNumber !== null}
+          label="Generate Invoice"
+          description="Invoice number will be automatically assigned"
+          isCompleted={generateInvoiceTask && reviewRequestTask}
+          // isCompleted={!!documentsUrl}
         >
-          <AssignNumberModal
-            modalTitle="Assign Invoice Number"
-            fieldName="Invoice number"
-            onAssign={handleAssignInvoiceNumber}
-          >
-            {invoiceNumber === null ? (
-              <Button
-                marginLeft="auto"
-                height="35px"
-                bg="background.gray"
-                _hover={{ bg: 'background.grayHover' }}
-                color="black"
-              >
-                <Text textStyle="xsmall-medium">Assign number</Text>
-              </Button>
-            ) : (
-              <Button variant="ghost" textDecoration="underline black">
-                <Text textStyle="caption" color="black">
-                  Edit number
-                </Text>
-              </Button>
-            )}
-          </AssignNumberModal>
+          {generateInvoiceTask && reviewRequestTask ? (
```

Comment:
```suggestion
          {generateInvoiceTask ? (
```
Same as above, unless the `reviewRequestTask` must also be `true`? In that case then there's something wrong with the logic.